envoy/server_test.go

```diff
@@ -53,47 +54,58 @@ func validateListener(serialisedListener *any.Any, svc service.Service) {
 
 	if svc.ProxyMode == "http" {
 		So(filters[0].GetName(), ShouldEqual, wellknown.HTTPConnectionManager)
-		// TODO: Switch to ptypes.MarshalAny when updating the Envoy API
-		// See the original implementation from 080e510
-		//nolint:staticcheck // ignore SA1019 for deprecated code
-		connectionManager, err := conversion.MessageToStruct(filters[0].GetConfig())
+		connectionManager := &hcm.HttpConnectionManager{}
+		err = ptypes.UnmarshalAny(filters[0].GetTypedConfig(), connectionManager)
 		So(err, ShouldBeNil)
-		So(connectionManager, ShouldNotBeZeroValue)
-		So(connectionManager.GetFields()["stat_prefix"].GetStringValue(), ShouldEqual, "ingress_http")
+		So(connectionManager.GetStatPrefix(), ShouldEqual, "ingress_http")
+		So(connectionManager.GetRouteConfig(), ShouldNotBeNil)
+		So(connectionManager.GetRouteConfig().GetVirtualHosts(), ShouldHaveLength, 1)
+		virtualHost := connectionManager.GetRouteConfig().GetVirtualHosts()[0]
+		So(virtualHost.GetName(), ShouldEqual, svc.Name)
```

Comment:
The virtual host name is set to `svc.Name`, as mentioned in https://github.com/Nitro/sidecar/pull/58.